doc/migration.md

```diff
@@ -0,0 +1,30 @@
+# Performing migrations #
+
+## Introduction ##
+
```

Comment:
Explain here that this documentation is targetting developers.

The best way is to make two main sections:

- Migration howto - overview
- Migration howto - advanced (for developers)

The overview will show upgrade/downgrade
The advance will include everything else